src/workload_type/server.rs

```diff
@@ -108,12 +113,27 @@ impl WorkloadType for ReplicatedServer {
         let mut resources = BTreeMap::new();
 
         let key = "deployment/".to_string() + self.kube_name().as_str();
-        let state = self.meta.deployment_status()?;
+        let state = self.meta.deployment_status().unwrap_or_else(|e| {
+            if e.to_string().contains("NotFound") {
```

Comment:
In what scenario will this happen? Could you clarify more?

Reply:
Thanks! We already detect in our status call. We check if the expected native object is present. I'm expanding to add the reconciliation action to add it back when it is not found. This could happen if the user accidentally deletes the object (which was the case for one of the customers trying the rudr). This fix was actually based on their feedback.